tick/stateful/functions.go

```diff
@@ -1354,6 +1355,27 @@ func (year) Signature() map[Domain]ast.ValueType {
 	return timeFuncSignature
 }
 
+type now struct {
+}
+
+func (now) Reset() {
+}
+
+// Return the current local time.
+func (now) Call(args ...interface{}) (v interface{}, err error) {
+	if len(args) != 0 {
+		return 0, errors.New("now expects exactly zero argument")
+	}
+
+	v = time.Now()
+
+	return
+}
+
+func (now) Signature() map[Domain]ast.ValueType {
+	return timeFuncSignature
```

Comment:
I think we want a slightly different function signature here.

The `timeFuncSignature` is

```
fn: ast.TTime -> ast.TInt
```

I think we want

```
fn: nil -> ast.TTime
```

Reply:
Ouch, you're right, I didn't pay attention to that bit. I'll rework on the PR and get back to you.